bigbluebutton-html5/imports/ui/components/settings/component.jsx

```diff
@@ -14,29 +14,56 @@ export default class Settings extends React.Component {
   constructor(props) {
     super(props);
     this.submenus = [];
+    this.state = { activeSubmenu: 0, focusSubmenu: 0 };
   }
 
-  componentWillMount() {
-    /* activeSubmenu represents the submenu in the submenus array to be displayed to the user,
-     * initialized to 0
-     */
-    this.setState({ activeSubmenu: 0 });
-    /* focusSubmenu represents the submenu in the submenus array which currently has focus,
-     * initialized to 0
-     */
-    this.setState({ focusSubmenu: 0 });
+  renderSettingOptions() {
+    const { isPresenter, role } = this.props;
+    this.submenus = [];
```

Comment:
You should move those `push's` to the `constructor` function. The render function should only contain render related logic.

And you can push multiple objects in a single push: `this.submenus.push({...}, {...}, {...}, ...)`